### PR TITLE
feat: QR code support, Milestone 1 - WPB-10529

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10529" title="WPB-10529" target="_blank">EpicWPB-10529</a>  QR in User Profile M1
  </td></table>
  <br />
 

### Issue

This feature branch is the result of the Hackathon work on QR code support. This feature was split into 2 milestones, and here is the scope of **milestone 1**:

Will contain adding the QR code button in the User profile - clicking on it will take user to Share profile - Share profile page will contain the QR code, Share profile link (which is the link which can be shared with in conversations) and Share QR code (which is the QR code which can be shared with in conversations).

[Design in Figma](https://www.figma.com/design/sOAbcd6UVtlZ5ejK4BoEvX/iOS)
Screenshot 2024-10-14 at 09 29 37

### Testing

User profile - QR code - Share the code

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        let controller = UIHostingController(rootView: view)
+        let targetSize = CGSize(width: 400, height: 400)
+        controller.view.bounds = CGRect(origin: .zero, size: targetSize)
+        controller.view.backgroundColor = .clear
+
+        let renderer = UIGraphicsImageRenderer(size: targetSize)
+        return renderer.image { _ in
+            controller.view.drawHierarchy(in: controller.view.bounds, afterScreenUpdates: true)
+        }
+    }
```
Comment (jullianm) [inline wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift:116]: regarding the issue with the QR code not showing up when sharing on iOS 16, this code fixes the issue, right ?

**Review by jullianm — APPROVED** [review]
👍

File: wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift (hunk 1 above)
Comment (KaterinaWire) [reply wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Views/QRCodeView.swift:116]: I've added 2 states:
      @State private var capturedImage: UIImage?
    @State private var shouldCaptureImage = false
    
For some reason, image wasn't captured at the moment we wanted to share in in iOS 16

**Review by agisilaos — APPROVED** [review]
Nice work! 🎉

**Pull request merged**